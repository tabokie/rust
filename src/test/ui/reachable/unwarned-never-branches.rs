// Test the behavior of `0 - 1` when overflow checks are disabled.
#![feature(never_type)]

#![deny(unreachable_code)]
#![allow(dead_code, invalid_value)]

enum Void {}

fn foo() -> Void {
    unsafe { std::mem::MaybeUninit::<Void>::uninit().assume_init() }
}

fn a() {
    if true {
        foo();
    }
    println!("I'm alive!");
}

fn b() {
    match true {
        true => {
            foo();
        }
        _ => {
            return;
        }
    }
    println!("I should be dead though."); // FIXME: false negative.
}

fn c() {
    match true {
        true => {
            foo();
            println!("I'm dead."); //~ ERROR unreachable expression
        }
        _ => {}
    }
}

fn main() {}
