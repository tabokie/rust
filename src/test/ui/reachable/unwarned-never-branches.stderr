error: unreachable expression
  --> $DIR/unwarned-never-branches.rs:36:13
   |
LL |             foo();
   |             ----- any code following this expression is unreachable
LL |             println!("I'm dead.");
   |             ^^^^^^^^^^^^^^^^^^^^^ unreachable expression
   |
note: the lint level is defined here
  --> $DIR/unwarned-never-branches.rs:4:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^
note: this expression has type `Void`, which is uninhabited
  --> $DIR/unwarned-never-branches.rs:35:13
   |
LL |             foo();
   |             ^^^^^
   = note: this error originates in the macro `$crate::format_args_nl` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

